streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -203,6 +203,10 @@ public String queryableStoreName() {
             processorSupplier,
             tableNode,
             builder);
+
+        kTable.enableSendingOldValues();
```

Comment:
Don't we need to call `this.enableSendingOldValues()` as "this" is the upstream KTable the filter is applied to? Why do we need to enable sending old values on the result of the filter() ?

Reply:
If we call `this.enableSendingOldValues()` then `enableSendingOldValues` is not called on the `KTableFilter` instance. Whereas calling `kTable.enableSendingOldValues()` does call `KTableFilter.enableSendingOldValues()`.

The call to `KTableFilter.enableSendingOldValues()` is needed to ensure the filter expects the old values.